internal/services/sources_service.go

```diff
@@ -9,16 +9,31 @@ import (
 )
 
 func ListSources(w http.ResponseWriter, r *http.Request) {
+	var sourcesList []*clients.Source
+
 	client, err := clients.GetSourcesClient(r.Context())
 	if err != nil {
 		renderError(w, r, payloads.NewClientError(r.Context(), err))
 		return
 	}
-	sourcesList, err := client.ListProvisioningSources(r.Context())
-	if err != nil {
-		renderError(w, r, payloads.NewClientError(r.Context(), err))
+	hyperscaler := r.URL.Query().Get("hyperscaler")
+	if hyperscaler == "gcp" || hyperscaler == "azure" || hyperscaler == "aws" {
```

Comment:
See my comment about clean API, here you could just use our own `ProviderType` type, then you can use existing function named `ProviderTypeFromString` to actually do this check in a more elegant way:

    if models.ProviderTypeFromString(param) != models.ProviderTypeUnknown {